nbs_generate_schematic.py

```diff
@@ -4,221 +4,239 @@
-    for instrument, singleChord in listOfChords.items():
-      lowerOctaveNotes = []
-      upperOctaveNotes = []
-      for note in singleChord:
-        if note.key < INSTRUMENT_RANGE[0] + 12:
-          lowerOctaveNotes.append(note)
-        else:
-          upperOctaveNotes.append(note)
-      if len(lowerOctaveNotes) > CHORD_MAX_SIZES[INSTRUMENTS[instrument]] or len(upperOctaveNotes) > CHORD_MAX_SIZES[INSTRUMENTS[instrument]]:
-        print('Warning: Your song contains chords that are larger than allowed.')
-        isValid = 0
-        break
-    if isValid == 0:
-      break
-
-  if isValid == 0:
-    sys.exit('We found some issues with your song. Please make sure to format it using the "nbs_format_song" script.')
-  else:
-    print('Song verified. Everything looks good!')
+
+def verifyFormat(song, songFile):
+    print('Verifying your song...')
+    isValid = True
+    # check song length
+    print('Checking song length...')
+    if song.header.song_length > MAX_SONG_LENGTH:
+        print('Warning: Your song is too long.')
+        isValid = False
+
+    # check custom instruments
+    print('Checking for custom instruments...')
+    if len(song.instruments) > 0:
+        print('Warning: Your song contains custom instruments.')
+        isValid = False
+
+    # check range
+    print('Checking note ranges...')
+    for note in song.notes:
+        if note.key < INSTRUMENT_RANGE[0] or note.key > INSTRUMENT_RANGE[1]:
+            print('Warning: Your song contains notes that are outside the normal range.')
+            isValid = False
+            break
+
+    # check chord lengths
+    print('Checking chord lengths...')
+    for tick, chord in song:
+        listOfChords = {}
+        for note in chord:
+            if note.instrument in listOfChords:
+                listOfChords[note.instrument].append(note)
+            else:
+                listOfChords[note.instrument] = [note]
+        for instrument, singleChord in listOfChords.items():
+            lowerOctaveNotes = []
+            upperOctaveNotes = []
+            for note in singleChord:
+                if note.key < INSTRUMENT_RANGE[0] + 12:
+                    lowerOctaveNotes.append(note)
+                else:
+                    upperOctaveNotes.append(note)
+            if len(lowerOctaveNotes) > CHORD_MAX_SIZES[INSTRUMENTS[instrument]] or len(upperOctaveNotes) > CHORD_MAX_SIZES[INSTRUMENTS[instrument]]:
+                print('Warning: Your song contains chords that are larger than allowed.')
+                isValid = False
+                break
+        if not isValid:
+            break
+
+    if not isValid:
+        input('We found some issues with your song. This is likely because you have not used the "nbs_format_song" script. Press Enter to automatically format your song.')
+        import nbs_format_song
+        nbs_format_song.main(songFile)
+        print("Reverifying your song...")
+        verifyFormat(song, songFile)
```

Comment:
Same as the comment on the Readme, I don't want to combine the two scripts so that the user can preview their song before a schematic is generated.